### ISSUE TITLE
please help me Excepcion cant Caught in laravel!

### ISSUE THREAD

**Issue by nikkotw** [mike42/escpos-php#957]
/**

 * PrintConnector for directly opening a network socket to a printer to send it commands.

 */

class NetworkPrintConnector extends FilePrintConnector

{

    /**

     * Construct a new NetworkPrintConnector

     *

     * @param string $ip IP address or hostname to use.

     * @param string $port The port number to connect on.

     * @param string $timeout The connection timeout, in seconds.

     * @throws Exception Where the socket cannot be opened.

     */

    public function __construct($ip, $port = "9100", $timeout = false)

    {

        // Default to 60 if default_socket_timeout isn't defined in the ini

        $defaultSocketTimeout = ini_get("default_socket_timeout") ?: 60;

        $timeout = $timeout ?: $defaultSocketTimeout;

 

        $this -> fp = @fsockopen($ip, $port, $errno, $errstr, $timeout);

        if ($this -> fp === false) {

            throw new Exception("Cannot initialise NetworkPrintConnector: " . $errstr);

        }

    }

}

Never Catch excepcion in laravel
Controlller

 function envia($dato){
        
        try {
            $connector = new NetworkPrintConnector("192.168.0.10", 9100);

            $printer = new Printer($connector);
            $printer->text($dato);
            $printer -> setJustification(Printer::JUSTIFY_RIGHT);
            $printer -> text("Default width\n");
            $printer -> setJustification(Printer::JUSTIFY_RIGHT);
            $printer -> text("Default width\n");
            $printer->cut();

            //Close printer 
            $printer->close();
            return 1;
        } catch(Exception $e) {
            return $e;
        } catch(Error $e) {
                return $e;
        } catch(Throwable $e) {
            return $e;
        }
}

never Return this excepction " Cannot initialise NetworkPrintConnector:"

**Comment by Mibrahimwali** [comment]
I am also having the same issue.. Have you found any solution?

**Comment by nikkotw** [comment]
Yes, I made an exception personalize, it's the only way at  least I think so

**Comment by dvalient** [comment]
Hello nikkotw, Could you explain a little more, how you have done the customization of the exception please
